From 4a57a869322416dddfeba5a6a54eb0b62ecce37a Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Mon, 18 Sep 2017 17:58:04 +0800
Subject: [PATCH] PM / devfreq: rockchip_dmc: Fix support for optional
 regulators

When not using the "_optional" function, a dummy regulator may be returned
and the driver fails to set voltage.

Change-Id: Ib636572362ece234d6b5694c8832ec01daa274e1
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/devfreq/rockchip_dmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/devfreq/rockchip_dmc.c b/drivers/devfreq/rockchip_dmc.c
index 65d4c9a2420a..63bbc96995c9 100644
--- a/drivers/devfreq/rockchip_dmc.c
+++ b/drivers/devfreq/rockchip_dmc.c
@@ -1533,7 +1533,7 @@ static int rockchip_dmcfreq_probe(struct platform_device *pdev)
 	mutex_init(&data->lock);
 	INIT_LIST_HEAD(&data->video_info_list);
 
-	data->vdd_center = devm_regulator_get(dev, "center");
+	data->vdd_center = devm_regulator_get_optional(dev, "center");
 	if (IS_ERR(data->vdd_center)) {
 		dev_err(dev, "Cannot get the regulator \"center\"\n");
 		return PTR_ERR(data->vdd_center);
-- 
2.35.3

